components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/EndpointsApiServiceImpl.java

```diff
@@ -242,6 +291,16 @@ public Response endpointsPost(EndPointDTO body, String contentType, String accep
             org.wso2.carbon.apimgt.rest.api.common.dto.ErrorDTO errorDTO = RestApiUtil.getErrorDTO(e.getErrorHandler());
             log.error(errorMessage, e);
             return Response.status(e.getErrorHandler().getHttpStatusCode()).entity(errorDTO).build();
+        } catch (JsonProcessingException e) {
```

Comment:
use catch clause as catch (JsonProcessingException | IOException e ) since you have the same error message.